pkg/backend/backend.go

```diff
@@ -74,6 +74,9 @@ type StackReference interface {
 	// but that information is not part of the StackName() we pass to the engine.
 	Name() tokens.Name
 
+	// Project is the project name that this stack belongs to. For old filestate backends this will be empty.
+	Project() tokens.Name
+
```

Comment:
One way we can guard against the whole "didn't handle empty case" in the future is making this return a tuple.
```suggestion
	// Project is the project name that this stack belongs to.
	// For old filestate backends this will return false.
	Project() (name tokens.Name, ok bool)

```

Reply:
Wouldn't have helped the place we last screwed this up (because it was inside the filestate internals so access the "project" field directly. But I'll add this to lessen the chance of us making that mistake anywhere else.